test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
-    },
-    { data: startingData, password },
-  );
-
-  // Update the data, encrypt using key
-  const newData = { ...startingData, bar: 'more data' };
-  const encryptWithKeyResult = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.keyString);
-      return await window.encryptor.encryptWithKey(key, args.data);
-    },
-    { data: newData, keyString: exportedKeyString },
-  );
 
-  // Mock the encrypted object
-  const decryptable = {
-    ...encryptWithKeyResult,
-    salt,
-  };
+      await expect(
+        page.evaluate(
+          async (args) => {
+            const key = await window.encryptor.keyFromPassword(
+              args.wrongPassword,
+              args.salt,
+            );
+            return await window.encryptor.decryptWithKey(
+              key,
+              args.encryptedPayload,
+            );
+          },
+          { encryptedPayload, salt, wrongPassword },
+        ),
+      ).rejects.toThrow('Incorrect password');
+    });
+
+    test('encryptor:decrypt encrypted data using key', async ({ page }) => {
+      const password = 'a sample passw0rd';
+      const expectedData = { foo: 'data to encrypt' };
+      const { salt } = testEncryptedData;
+
+      const decryptedData = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.keyFromPassword(
+            args.password,
+            args.salt as string,
+            false,
+            args.testEncryptedData.keyMetadata,
+          );
+          return await window.encryptor.decryptWithKey(
+            key,
+            args.testEncryptedData,
+          );
+        },
+        { testEncryptedData, password, salt },
+      );
 
-  // Prove that a vault created with key can be decrypted with password
-  const decryptedResult = await page.evaluate(
-    async (args) =>
-      await window.encryptor.decryptWithDetail(args.password, args.data),
-    { password, data: JSON.stringify(decryptable) },
-  );
+      expect(decryptedData).toStrictEqual(expectedData);
+    });
+
+    test('encryptor:decrypt encrypted data using key derived from wrong password', async ({
+      page,
+    }) => {
+      const wrongPassword = 'a wrong password';
+
+      await expect(
+        page.evaluate(
+          async (args) => {
+            const key = await window.encryptor.keyFromPassword(
+              args.wrongPassword,
+              args.salt as string,
+              false,
+              args.encryptedPayload.keyMetadata,
+            );
+            return await window.encryptor.decryptWithKey(
+              key,
+              args.encryptedPayload,
+            );
+          },
+          {
+            encryptedPayload: testEncryptedData,
+            salt: testEncryptedData.salt,
+            wrongPassword,
+          },
+        ),
+      ).rejects.toThrow('Incorrect password');
+    });
+
+    test('encryptor:importKey generates valid CryptoKey', async ({ page }) => {
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop? (but see possible comment from Nicholas about this)